### PR TITLE
Require --force-confold for dpkg upgrades

### PR DESCRIPTION
**Pull request by jxmx**
Unconditionally set `--force-confold` for dpkg operations

### CODE REVIEW

**Review by Allan-N — COMMENTED** [review]
Q? does this change apply to ALL configuration files that are updated?  or is it limited to just the ASL3 packages.  I'm a bit hesitant if it's the former.

**Comment by jxmx** [comment]
It would be universal.

**Comment by encbar5** [comment]
Maybe providing guidance in the manual would be preferable?
Like, to do an upgrade, 
1. backup configuration files (can provide commands or asl-menu)
2. sudo apt update && sudo apt upgrade
3. if needed, the config files can be restored from the backup

Or
1. sudo apt update
2. sudo apt upgrade asl3 --force-confold (not sure if this is correct, but just an explicit command to only apply this option to asl3)
3. sudo apt upgrade (now upgrade the rest of the system)

Any of the above could be put in to a "safe_upgrade" script as well, if that helps users.

Just throwing these ideas out there.

**Comment by jxmx** [comment]
We have language in there now of always picking "No" and people still do it. Getting people to follow alternative directions for this seems particularly challenging.

**Comment by Allan-N** [comment]
Need something that will "do the right thing" for all packages and handle the case of "non-interactive / no interaction", "Cockpit", and command-line.

Hmmm...  I wonder if adding a new "asl3-update" command would help?

**Comment by Allan-N** [comment]
Q? is it possible add pre/post upgrade scripts to our package(s) that could copy and restore the current configuration files?

**Comment by jxmx** [comment]
I would really like to avoid creating a custom command to do updates because all standard Debian tools - normal apt calls, Cockpit, unattended-upgrades, etc. - won't use it. And, fundamentally, I think we should adhere as closely as possible to "standard debian" so that as ASL progresses, it always "just works" with all of the standard tooling people expect.

It is always possible to revert the config to pristine with `apt install -o Dpkg::Options::="--force-confnew" --reinstall <package-name>`.

Part of the problem (tagging @encbar5 ) is that we already have information in the manual on upgrading and always choosing "N" on package installs but people don't read it. Adding even MORE documentation on how to update will never be read by the people that actually need the info.

My solution of conditionally keeping all old config files is actually the reasonable default for any intra-release upgrades. Moving from Debian 12 to 13 might be a different story but that's not a bridge we're ready to cross yet.

File: apt.conf.d/Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+sysconfdir ?= /etc/apt.conf.d/
```
Comment (Allan-N) [inline apt.conf.d/Makefile]: Should this path be `/etc/apt/apt.conf.d` ?

**Review by Allan-N — CHANGES_REQUESTED** [review]
OK.  I've been convinced that this path is acceptable.

Aside from the /etc/... path question my only suggestion would be to change ".../apt.conf.d/99asl-confold" to a more descriptive ".../apt.conf.d/99asl-keep-current-configurations".

File: apt.conf.d/Makefile (hunk 1 above)
Comment (jxmx) [reply apt.conf.d/Makefile]: Good catch.

**Review by Allan-N — APPROVED** [review]

**Pull request merged**